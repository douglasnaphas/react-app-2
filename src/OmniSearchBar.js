import React, { useState } from "react";
import PropTypes from "prop-types";

const OmniSearchBar = ({ onSubmitSearch }) => {
  const [query, setQuery] = useState("");

  const valueChanged = evt => {
    console.log("value changed: " + query);
    setQuery(evt.target.value);
  };

  const enterPressed = evt => {
    const code = evt.keyCode || evt.which;
    console.log("enterPressed: " + query);
    if (code === 13 && query.length > 2) {
      onSubmitSearch(query);
    }
  };

  console.log("render: " + query);
  return (
    <div className="omni-search-bar">
      <tr>
        <input
          type="search"
          placeholder="across all accounts"
          className="search-bar"
          onChange={valueChanged}
          value={query}
          onKeyPress={enterPressed}
        />
      </tr>
    </div>
  );
};

OmniSearchBar.propTypes = {
  onSubmitSearch: PropTypes.func.isRequired
};

export default OmniSearchBar;
