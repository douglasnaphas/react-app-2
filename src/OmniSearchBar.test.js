import React from "react";
import OmniSearchBar from "./OmniSearchBar";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("OmniSearch/OmniSearchPage", () => {
  let onSubmitFunc;

  beforeEach(() => {
    onSubmitFunc = jest.fn();
  });

  it("three characters are required to initiate search", () => {
    const component = shallow(<OmniSearchBar onSubmitSearch={onSubmitFunc} />);
    let input = component.find(".search-bar");

    expect(input.exists()).toEqual(true);
    expect(component).toMatchSnapshot();

    input.simulate("keypress", { keyCode: 13 });
    expect(onSubmitFunc).not.toHaveBeenCalled();

    input = component.find(".search-bar");
    input.prop("onChange")({ target: { value: "te" } });
    expect(component).toMatchSnapshot();
    input.simulate("keypress", { keyCode: 13 });
    expect(onSubmitFunc).not.toHaveBeenCalled();

    input = component.find(".search-bar");
    input.prop("onChange")({ target: { value: "test1234" } });
    expect(component).toMatchSnapshot();
    input = component.find(".search-bar");
    input.simulate("keypress", { keyCode: 13 });
    input = component.find(".search-bar");
    expect(onSubmitFunc).toHaveBeenCalled();
  });
});
