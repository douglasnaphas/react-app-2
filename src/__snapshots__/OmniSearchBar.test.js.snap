// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OmniSearch/OmniSearchPage three characters are required to initiate search 1`] = `ShallowWrapper {}`;

exports[`OmniSearch/OmniSearchPage three characters are required to initiate search 2`] = `ShallowWrapper {}`;

exports[`OmniSearch/OmniSearchPage three characters are required to initiate search 3`] = `ShallowWrapper {}`;
